processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/InheritanceSelector.java

```diff
@@ -41,11 +42,11 @@
                                                                           Type targetType,
                                                                           SelectionCriteria criteria) {
 
-        if ( sourceTypes.size() != 1 ) {
+        if ( sourceTypes.size() != 1 && criteria.getAssignment() == null) {
```

Comment:
I am not entirely sure if we need this here, maybe it is wrong. I think it works without that as well. 

On top of that. This together with #1242 will lead to a runtime error on line 56.